src/core.h

```diff
@@ -96,6 +97,14 @@ struct GroupedValues {
  */
 const char* get_version();
 
+/** Return a reference to the global PRNG. */
+static inline std::mt19937& prng()
+{
+    static std::mt19937 PRNG;
```

Comment:
The default random engine is system-specific so not very reproducible (and usually crappy).
MT199937 is good default choice (good property, good performance).

This version only generate 32-bit numbers (like the `rand` that was used before), but it can be modified to return 64-bit or double or if we want.
I kept 32-bit for compatibility with what was done so far and because some tests doesn't work with number bigger than 32-bit (as I explained in the commit message, I guess that the primality test is too slow to handle 64-bit numbers).